Sources/MongoSwift/Operations/ChangeStreamNextOperation.swift

```diff
@@ -0,0 +1,48 @@
+import mongoc
+
+/// An operation corresponding to a `next` call on a ChangeStream.
+internal struct ChangeStreamNextOperation<T: Codable>: Operation {
+    private let changeStream: ChangeStream<T>
+
+    internal init(changeStream: ChangeStream<T>) {
+        self.changeStream = changeStream
+    }
+
+    internal func execute(using connection: Connection, session: ClientSession?) throws -> T? {
+        // If an error exists, refuse iterating the change stream to avoid overwriting the original error.
+        guard self.changeStream.error == nil else {
```

Comment:
we already check this in `next`, just before constructing the operation, so can probably skip it here.